### DISCUSSION TITLE
Constant disk I/O writes from LND

### DISCUSSION THREAD

**Discussion by guitarchair** [lightningnetwork/lnd#8929] [Troubleshooting] [answered]
My SSD is blinking non-stop since installing LND. I used the iotop command to see what's causing it. I see lnd at the top all the time, nonstop:
umbrel      0.00 B/s  128.23 K/s lnd --configfile=/data/.lnd/umbrel-lnd.conf --listen=0.0.0.0:9~1.9 --tor.password=

The lnd is constantly writing to disk. This will damage the disk eventually, running 24/7
The lnd logs seem okay, only once a minute there's a new line:
2024-07-23 20:05:47.683 [INF] CRTR: Processed channels=0 updates=100 nodes=24 in last 59.99930507s
2024-07-23 20:06:47.684 [INF] CRTR: Processed channels=0 updates=109 nodes=19 in last 1m0.000261277s
2024-07-23 20:07:47.684 [INF] CRTR: Processed channels=0 updates=86 nodes=41 in last 1m0.000116687s
2024-07-23 20:08:47.683 [INF] CRTR: Processed channels=0 updates=93 nodes=29 in last 59.999173829s

This issue seems related but it's from 4 years ago and already closed:
https://github.com/lightningnetwork/lnd/issues/4689

The node itself is working fine and I can send lightning payments. But this constant disk write needs to be fixed.
I'm using the latest lnd v0.18.0-beta

**Comment by guggero** [comment]
`lnd` is maintaining a network graph of a p2p network which is communicated through gossip messages. Those messages come in all the time and for you to be able to use the network you'll want to persist that data.
So "this constant disk write needs to be fixed" isn't going to be possible, unless you completely turn off graph syncing.

But you can reduce the interval in which that happens (which has its own set of tradeoffs):
```
      --db.batch-commit-interval=                            The maximum duration the channel graph batch schedulers will wait before attempting to commit
                                                             a batch of pending updates. This can be tradeoff database contenion for commit latency.
                                                             (default: 500ms)
```

**Reply by guitarchair** [reply]
Thank you for the answer! I have a few follow-up questions:
Can you please elaborate on the trade off? What is the downside if I increase it to once a minute? And what's a safe number I can increase it to?
My situation is that I rarely use the node for LN payments (once a week) and have only one private channel open...
And I'm thinking maybe this data should be saved in RAM for longer, before writing to disk...?

EDIT: According to this link I should add the option to lnd.conf file:
https://github.com/lightningnetwork/lnd/blob/master/sample-lnd.conf

**Reply by guggero** [reply, marked as answer]
Once a minute should be fine. The tradeoff is basically how much data you lose if your node crashes unexpectedly. So I wouldn't really use a value greater than 5 minutes.

> And I'm thinking maybe this data should be saved in RAM for longer, before writing to disk...?

That's exactly what's happening, the `db.batch-commit-interval` dictates how often it's written from RAM to disk.

> According to this link I should add the option to lnd.conf file:

Yes, you'll need to add `db.batch-commit-interval=1m` to the `lnd.conf` file in Umbrel (not sure how exactly you do that, as I'm not very familiar with Umbrel).

The leading dashes (`--`, mentioned above) are only needed if you specify the option as a command line argument, so don't include those in the config file.

**Reply by guitarchair** [reply]
Thank you for answering! I'll try to do that

**Reply by olafmarcos** [reply]
I tested to change the config db.batch-commit-interval to 1 minute and the constants writes to disk continues.

    PID  PRIO  USER     DISK READ DISK WRITE>    COMMAND
 819910 be/4 xxxx      0.00 B/s    3.41 K/s lnd --configfile=/data/.lnd/umbrel-lnd.conf --listen=0.0.0.0:9735 --rpc~=10.21.21.9 --tor.password=xxxx

Regards,
